addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2628,7 +2693,11 @@ const SnippetOptionWidget = Widget.extend({
             value = value.split(params.saveUnit).join('');
         }
         if (params.extraClass) {
-            this.$target.toggleClass(params.extraClass, params.defaultValue !== value);
+            if (params.defaultValue === value) {
+                await this.editorHelpers.removeClass(context, this.$target[0], params.extraClass, context);
+            } else {
+                await this.editorHelpers.addClass(context, this.$target[0], params.extraClass, context);
+            }
```

Comment:
This is not a toggle. You want the class gone in one case, whether it was there to begin with or not, and you want it there in the other, whether it was there to begin with or not. Toggling would mean that if it was there already, we'd remove it even though we want to keep it.
For `context` I'll direct you to @Goaman but simply put, here, it's the editor instance.

Reply:
I can read that... :wink: If you don't like the "toggle" word, then use another but a parameter to say if you want it "on" or "off" would just be normal for a JavaScript developer (https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle), rather than transforming a one liner into 5 lines for no reason.

Reply:
Sure we could have a helper for this I have no issue with that. I ddin't know about `DOMTokenList/toggle` so TIL... ;-) I still definitely wouldn't call it `toggle` but that's me be a semantics extremist ^^
In any case you'll get your helper :-D

Reply:
Maybe should have mentioned that: `DOMTokenList` being the `Class` used for the `classList` instance so it's not a random unused class :wink:

Reply:
The `context` exists because the editor has to understand what is one batch of changes.

Here, the `this.editorHelpers.addClass` or `this.editorHelpers.removeClass` should be in the same batch as the caller of `_selectAttributeHelper`. The batch will be the same as the one of the caller.

Reply:
The helper is done, it's called `setClass`, will be merged soon.
Seen in context it's pretty neat and satisfies my semantics ocd ^^ :
```js
await this.editorHelpers.setClass(context, this.$target[0], 'o_bg_img_opt_repeat', widgetValue === 'repeat-pattern');
await this.editorHelpers.setStyle(context, this.$target[0], 'background-position', '');
```

Reply:
I'm closing this comment since @Zinston's `setClass` helper was merged today.